File: TriCoachAI/Core/Networking/APIConfig.swift
import Foundation

enum APIConfig {
    /// Branches on simulator-vs-device rather than Debug-vs-Release:
    /// `localhost` is exempt from App Transport Security and, in the
    /// Simulator, resolves to the Mac itself (sharing its network stack),
    /// so it reaches `npm run dev` with no Info.plist exceptions. On a real
    /// device — even a Debug build, which is Xcode's default "Run"
    /// configuration — `localhost` would resolve to the device's own
    /// loopback instead and silently fail to reach anything, so devices
    /// always use the deployed backend.
    static let baseURL: URL = {
        #if targetEnvironment(simulator)
        return URL(string: "http://localhost:3000/api/v1")!
        #else
        return URL(string: "https://tricoach-9ob8.onrender.com/api/v1")!
        #endif
    }()
}

File: TriCoachAndroid/app/src/main/java/com/tricoach/android/core/network/ApiConfig.kt
package com.tricoach.android.core.network

import com.tricoach.android.BuildConfig

/** Debug builds target the local dev backend via the emulator's host-loopback alias (10.0.2.2 = the host machine's localhost) — the Android analog of iOS's simulator-vs-device APIConfig branch. */
object ApiConfig {
    val baseUrl: String =
        if (BuildConfig.DEBUG) "http://10.0.2.2:3000/api/v1/" else "https://tricoach-9ob8.onrender.com/api/v1/"
}
